indra/newview/llviewermenu.cpp

```diff
@@ -6913,6 +6914,59 @@ void handle_give_money_dialog()
     }
 }
 
+LLFloaterSidePanelContainer* get_favorite_inventory_folder()
+{
+    LLUUID folder_id = LLUUID(gSavedPerAccountSettings.getString("FavoriteFolder"));
+    if (!folder_id.isNull())
+    {
+        LLFloaterReg::const_instance_list_t& inst_list = LLFloaterReg::getFloaterList("inventory");
+        for (LLFloaterReg::const_instance_list_t::const_iterator iter = inst_list.begin(); iter != inst_list.end();)
+        {
+            LLFloaterSidePanelContainer* inventory_container = dynamic_cast<LLFloaterSidePanelContainer*>(*iter++);
+            if (inventory_container)
+            {
+                LLSidepanelInventory* sidepanel_inventory = dynamic_cast<LLSidepanelInventory*>(inventory_container->findChild<LLPanel>("main_panel", true));
+                if (sidepanel_inventory)
+                {
+                    LLPanelMainInventory* main_inventory = sidepanel_inventory->getMainInventoryPanel();
+                    if (main_inventory && main_inventory->isSingleFolderMode()
+                        && (main_inventory->getCurrentSFVRoot() == folder_id))
+                    {
+                        return inventory_container;
+                    }
+                }
+            }
+        }
+    }
+
+    return NULL;
+}
+
+void toggle_favorite_inventory_folder()
+{
+    LLUUID folder_id = LLUUID(gSavedPerAccountSettings.getString("FavoriteFolder"));
+    if ((folder_id.isNull()) || (!gInventory.getCategory(folder_id)))
+    {
+        LLNotificationsUtil::add("MissingFavoriteFolder");
+        return;
+    }
+
+    LLFloaterSidePanelContainer* inventory_container = get_favorite_inventory_folder();
+    if (inventory_container)
+    {
+        inventory_container->closeFloater();
+    }
+    else
+    {
+        LLPanelMainInventory::newFolderWindow(folder_id);
+    }
+}
+
+bool favorite_inventory_folder_visible()
+{
+    return (get_favorite_inventory_folder() != NULL);
```

Comment:
not tested how often this is polled, most of the toolbar buttons use `LLFloater::isVisible` so I'm assuming this is probably absolutely fine